optimade/filtertransformers/json.py

```diff
@@ -7,7 +7,7 @@ def __init__(self, compact=False):
         self.compact = compact
         super().__init__()
 
-    def __default__(self, data, children):
+    def __default__(self, data, children, meta):
```

Comment:
I guess this is not for `lark` version 0.5.6 but a newer version?

Reply:
Commenting here as it reminded me: we fail a test with the latest lark version (that gets installed by default), some issue with a linebreak in a test file (so probably harmless). We should probably check this and update to the latest lark as part of this PR.

Reply:
Yes, sorry, I had random fails on my mac when I tried to the test and probably I messed up my conda env.

Reply:
If nobody against it, I would try to upgrade the version of `lark-parser` (latest: 0.7.7) and fix the issue with the [test](https://github.com/Materials-Consortia/optimade-python-tools/blob/filter_v0.10.0/optimade/filterparser/tests/testfiles/parse_012.inp). Please correct me, but this test should fail anyway because the `\n` shouldn't be allowed in the string of the filter, should it?

Reply:
@ml-evs Unfortunately, there is some issue with the tests after updating the version of `lark-parser`. The previous implementation of the grammar completely ignored the [precedence](https://github.com/Materials-Consortia/OPTiMaDe/blob/master/optimade.md#precedence) between AND and OR and those tests can randomly fail.

Reply:
I had to use parenthesis in the test to get rid of the ambiguity of the grammar

Reply:
What is the conclusion of updating to the latest Lark version concerning the tests?